ui/components/DesignLifeCycle/SelectDeploymentTarget.js

```diff
@@ -56,9 +56,10 @@ const K8sContextConnection = ({ connection, environment }) => {
   );
   const dispatch = useDispatchRtk();
   const toggleK8sConnection = () => dispatch(toggleConnection(environment, connection));
+
   return (
     <K8sContextConnectionChip
-      ctx={connection}
+      ctx={{ ...connection.metadata, connection_id: connection.id }}
```

Comment:
we should directly use the whole connection as the prop . this will otherwise make it confusing to know what is included in ctx

Reply:
and also rename the prop to connection than ctx .

Reply:
The thing is, if I pass the whole connection it doesn't set the connection_id. And inside deployment it fails.
If we do this it specifically sets the connection_id
` ctx={{ ...connection, connection_id: connection.id }}`

Reply:
the issue is connetion.metadata.id is getting intermixed with connection.id  ,